File: app/src/main/java/com/ayat/candyapp/login/UserManagementRepository.java
package com.ayat.candyapp.login;

import com.blessedtreeit.product.activitycenter.UserPreferences;
import com.blessedtreeit.product.activitycenter.flows.login.model.LoginModel;
import com.blessedtreeit.product.activitycenter.flows.login.model.LoginRequestModel;
import com.blessedtreeit.product.activitycenter.network.EndPoint;
import io.reactivex.Single;

import javax.inject.Inject;
import javax.inject.Singleton;

/**
 * Created by Ayat khraisat  on 12/27/18
 * akhraisat@blessedtreeit.com
 * <p>
 * Project Name: ActivityCenter
 * <p>
 * Blessed Tree IT
 */

@Singleton
public class UserManagementRepository {

    private UserPreferences userPreferences;
    private EndPoint endPoint;

    @Inject
    public UserManagementRepository(EndPoint endPoint, UserPreferences userPreferences) {
        this.endPoint = endPoint;
        this.userPreferences = userPreferences;
    }

    public Single<LoginModel> getLoginObservable(String userName, String password) {
        return endPoint.login(
                new LoginRequestModel(userName, password, "", ""))
                .doOnSuccess(loginModel -> {
                    if (loginModel.isSuccess()) {
                        setUserName(loginModel.getFullName());
                    }
                });
    }

    public String getUserName() {
        return userPreferences.getUserName();
    }

    public void setUserName(String name) {
        userPreferences.setUserName(name);
    }

    public void setIsLoggedIn(boolean isLoggedIn) {
        userPreferences.setLoggedIn(isLoggedIn);
    }

    public boolean getIsLoggedIn() {
        return userPreferences.isLoggedIn();
    }



}

File: app/src/main/java/com/ayat/candyapp/login/UserManagementRepository.kt
package com.ayat.candyapp.login


import com.ayat.candyapp.UserPreferences
import com.ayat.candyapp.bases.BaseResponse
import com.ayat.candyapp.login.model.LoginModels
import com.ayat.candyapp.network.EndPoint
import com.ayat.candyapp.signup.models.SignUpRequestModel
import kotlinx.coroutines.Deferred

import javax.inject.Inject
import javax.inject.Singleton

/**
 *Created by Ayat Khriasat on 26,April,2019 at 10:57 PM
 *Email: ayatzkhraisat@gmail.com
 *Project: CandyApp
 **/
@Singleton
class UserManagementRepository @Inject
constructor(private val endPoint: EndPoint, private val userPreferences: UserPreferences) {

    var userName: String
        get() = userPreferences.userName
        set(name) {
            userPreferences.userName = name
        }

    var isLoggedIn: Boolean
        get() = userPreferences.isLoggedIn
        set(isLoggedIn) {
            userPreferences.isLoggedIn = isLoggedIn
        }

    fun getLoginDeferred(userName: String, password: String): Deferred<LoginModels.LoginResponseModel> {
        return endPoint.loginAsync(
            LoginModels.LoginRequestModel(userName, password)
        );
    }

    fun getSiguoDeffered(userName: String, password: String): Deferred<BaseResponse> {
        return endPoint.signUpAsync(
            SignUpRequestModel(userName, password)
        );
    }


}
